ext/hook/uhook_attributes.c

```diff
@@ -0,0 +1,432 @@
+                    zval_ptr_dtor(&value);
+                    continue;
+                }
+            }
+
+            if (zend_string_equals_literal(name, "name") && Z_TYPE(value) == IS_STRING) {
+                if (!def->name) {
+                    def->name = zend_string_copy(Z_STR(value));
+                }
+            } else if (zend_string_equals_literal(name, "resource") && Z_TYPE(value) == IS_STRING) {
+                if (!def->resource) {
+                    def->resource = zend_string_copy(Z_STR(value));
+                }
+            } else if (zend_string_equals_literal(name, "service") && Z_TYPE(value) == IS_STRING) {
+                if (!def->service) {
+                    def->service = zend_string_copy(Z_STR(value));
+                }
+            } else if (zend_string_equals_literal(name, "type") && Z_TYPE(value) == IS_STRING) {
+                if (!def->type) {
+                    def->type = zend_string_copy(Z_STR(value));
+                }
+            } else if (zend_string_equals_literal(name, "args")) {
+                if (!def->arg_whitelist) {
+                    if (Z_TYPE(value) == IS_ARRAY) {
+                        def->args = true;
+                        def->arg_whitelist = zend_new_array(zend_hash_num_elements(Z_ARR(value)));
+                        zval *val;
+                        ZEND_HASH_FOREACH_VAL(Z_ARR(value), val) {
+                            if (Z_TYPE_P(val) == IS_STRING) {
+                                zend_hash_add_empty_element(def->arg_whitelist, Z_STR_P(val));
+                            }
+                        } ZEND_HASH_FOREACH_END();
+                    } else {
+                        def->args = zend_is_true(&value);
+                    }
+                }
+            } else if (zend_string_equals_literal(name, "return")) {
+                def->retval = zend_is_true(&value);
+            } else if (zend_string_equals_literal(name, "run_if_limited")) {
+                def->run_if_limited = zend_is_true(&value);
+            } else if (zend_string_equals_literal(name, "recurse")) {
+                def->disallow_recursion = !zend_is_true(&value);
+            } else if (zend_string_equals_literal(name, "tags") && Z_TYPE(value) == IS_ARRAY) {
+                if (!def->tags) {
+                    def->tags = Z_ARR(value);
+                    GC_ADDREF(def->tags);
+                }
+            }
+
+            zval_ptr_dtor(&value);
+        }
+
+        zai_hook_install_resolved_generator(func,
+                                            dd_uhook_begin, dd_uhook_generator_resumption, dd_uhook_generator_yield, dd_uhook_end,
+                                            ZAI_HOOK_AUX(def, dd_uhook_dtor), sizeof(dd_uhook_dynamic));
+
+    }
+}
+
+void zai_uhook_attributes_minit(void) {
+    zai_hook_on_function_resolve = dd_uhook_on_function_resolve;
+
+    ddtrace_hook_attribute_ce = register_class_DDTrace_Traced();
+#if PHP_VERSION_ID >= 80200
```

Comment:
For posterity, shouldn't you try to explain in comments why the special cases like this one? Or will it be obvious to any PHP dev?

Reply:
The method does not exist yet - that's the only reason, which is immediately obvious when commenting the #if out :-D